core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -0,0 +1,205 @@
+  private[kafka] case object Watch
+  private[kafka] case object Stop
+  private[kafka] case class Commit(offsets: Map[TopicPartition, Long])
+  //responses
+  private[kafka] case class Assigned(partition: List[TopicPartition])
+  private[kafka] case class Revoked(partition: List[TopicPartition])
+  private[kafka] case class Messages[K, V](messages: Iterator[ConsumerRecord[K, V]])
+  private[kafka] case class Committed(offsets: Map[TopicPartition, OffsetAndMetadata])
+  //internal
+  private case object Poll
+
+  private var number = 0
+  private[kafka] def nextNumber() = {
+    this.synchronized {
+      number += 1
+      number
+    }
+  }
+  private[kafka] def rebalanceListener(onAssign: Iterable[TopicPartition] => Unit, onRevoke: Iterable[TopicPartition] => Unit) = new ConsumerRebalanceListener {
+    override def onPartitionsAssigned(partitions: util.Collection[TopicPartition]): Unit = {
+      onAssign(partitions.asScala)
+    }
+    override def onPartitionsRevoked(partitions: util.Collection[TopicPartition]): Unit = {
+      onRevoke(partitions.asScala)
+    }
+  }
+}
+
+private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V]) extends Actor {
+  import KafkaConsumerActor._
+
+  def pollTimeout() = settings.pollTimeout
+  def pollInterval() = settings.pollInterval
+
+  var requests = Map.empty[TopicPartition, ActorRef]
+  var consumer: KafkaConsumer[K, V] = _
+  var finished: Promise[Done] = _
+  var nextScheduledPoll: Option[Cancellable] = None
+  var pollExpected = false
+  var commitsInProgress = 0
+  var stopInProgress = false
+
+  def receive: Receive = LoggingReceive {
+    case Watch => sender() ! finished.future
+    case Assign(tps) =>
+      val previousAssigned = consumer.assignment()
+      consumer.assign((tps.toSeq ++ previousAssigned.asScala).asJava)
+    case AssignWithOffset(tps) =>
+      consumer.assign(tps.keys.toSeq.asJava)
+      tps.foreach {
+        case (tp, offset) => consumer.seek(tp, offset)
+      }
+    case Commit(offsets) =>
+      val commitMap = offsets.mapValues(new OffsetAndMetadata(_))
+      val reply = sender()
+      commitsInProgress += 1
+      consumer.commitAsync(commitMap.asJava, new OffsetCommitCallback {
+        override def onComplete(offsets: util.Map[TopicPartition, OffsetAndMetadata], exception: Exception): Unit = {
+          commitsInProgress -= 1
+          if (exception != null) reply ! Status.Failure(exception)
+          else reply ! Committed(offsets.asScala.toMap)
+        }
+      })
+      //right now we can not store commits in consumer - https://issues.apache.org/jira/browse/KAFKA-3412
+      pollExpected = true
+      poll()
+    case Subscribe(topics, listener) =>
+      consumer.subscribe(topics.toList.asJava, listener)
+    case SubscribePattern(pattern, listener) =>
+      consumer.subscribe(Pattern.compile(pattern), listener)
+    case Poll =>
+      pollExpected = true
+      poll()
+    case RequestMessages(topics) =>
```

Comment:
I guess it depends on whether these requests cause resources to leak or being held for overly long time if the requester goes away silently. If this is not an issue, then no need for watch.